mcp-auth/python/README.md

```diff
@@ -0,0 +1,160 @@
+- FastMCP framework with built-in authentication middleware
+- Custom JWTTokenVerifier implementation
+- Weather service with `get_weather` tool that requires authentication
+
+## Prerequisites
+
+- Python 3.12 or higher
+- Asgardeo account and application setup
+- pip (Python package installer)
+
+## Project Structure
+
+```
+├── main.py              # Main FastMCP server application
+├── jwt_validator.py     # JWT validation module
+├── README.md           # This file
+└── requirements.txt    # Python dependencies
+```
+
+## Installation
+
+1. **Clone or download this project**
+
+2. **Create a virtual environment (recommended)**
+   ```bash
+   python -m venv .venv
+   source .venv/bin/activate  # On Windows: .venv\Scripts\activate
+   ```
+
+3. **Install required dependencies**
+   ```bash
+   pip install -r requirements.txt
+   ```
+
+## Asgardeo Configuration
+
+### 1. Create an Asgardeo Application
+
+1. Login to your Asgardeo account.
+2. Navigate to the Applications Tab and select the **MCP Client Application** as shown in below image.
+![mcp-client-app-selection.png](resources/mcp-client-app-selection.png)
+
+3. Add your application name and callback URL
+![mcp-client-creation.png](resources/mcp-client-creation.png)
+
+### 2. Get Your Application Credentials
+
+Once the application is created get both the **Client ID** and **Tenant Name**:
+- **Client ID**: Found in the application's **Protocol** tab
+- **Tenant Name**: Your organization's tenant name (visible in the URL)
+
+### 3. Configure the Application
+
+Edit `main.py` and update the following variables:
+
+```python
+# Replace 'your-tenant' with your actual Asgardeo tenant name
+AUTH_ISSUER = "https://api.asgardeo.io/t/<your-tenant>/oauth2/token"
+CLIENT_ID = "your-client-id-here"  # Replace with your Asgardeo OAuth2 client ID
+JWKS_URL = "https://api.asgardeo.io/t/<your-tenant>/oauth2/jwks"
+```
+
+**Example with actual values:**
+```python
+AUTH_ISSUER = "https://api.asgardeo.io/t/mycompany/oauth2/token"
+CLIENT_ID = "abc123xyz789_client_id_from_asgardeo"
+JWKS_URL = "https://api.asgardeo.io/t/mycompany/oauth2/jwks"
+```
+
+Also update the AuthSettings URLs:
+```python
+auth=AuthSettings(
+    issuer_url=AnyHttpUrl("https://api.asgardeo.io/t/<your-tenant>/oauth2/token"),
+    resource_server_url=AnyHttpUrl("http://localhost:8000"),
+    # required_scopes=["user"]  # Uncomment and modify if you need specific scopes
+),
+```
+
+## Running the Server
+
+### Development Mode
+
+1. **Start the server**
+   ```bash
+   python main.py
+   ```
+
+2. **Server will start on**: `http://localhost:8000` using streamable-http transport
+
+
+## Test with MCP Inspector
+** Note **: Currently MCP Inspector doesn't support getting the authorization server information with `.well-known/openid-configuration`. Therefore, we are going to use a customized inspector that is supported. Please get it from here: [MCP Inspector](https://github.com/shashimalcse/inspector/tree/fix-auth)
```

Comment:
the latest inspector now supports falling back to OIDC discovery (just tested a few minutes ago). So, we wouldn't need to point to our fork now.